ArcGISRuntimeSDKQt_QMLSamples/Scenes/RealisticLightingAndShadows/RealisticLightingAndShadows.qml

```diff
@@ -0,0 +1,151 @@
+                // add an arcgis tiled elevation source...elevation source is a default property of surface
+                ArcGISTiledElevationSource {
+                    url: "https://elevation3d.arcgis.com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer"
+                }
+            }
+
+            //! [add a scene service with ArcGISSceneLayer]
+            ArcGISSceneLayer {
+                url: "https://tiles.arcgis.com/tiles/P3ePLMYs2RVChkJx/arcgis/rest/services/DevA_BuildingShells/SceneServer/layers/0"
+            }
+        }
+    }
+
+    Rectangle {
+        anchors {
+            top: parent.top
+            right: parent.right
+            margins: 5
+        }
+        width: childrenRect.width
+        height: childrenRect.height
+        color: "#ffffff"
+        opacity: 0.7
+        Column {
+            id: settingsColumn
+            spacing: 5
+            padding: 15
+            Text {
+                id: timeValue
+                anchors.horizontalCenter: parent.horizontalCenter
+                font.pixelSize: 16
+                text: "August 10, 2018 " + sceneView.sunTime.toLocaleTimeString("hh:mm ap")
+            }
+
+            ComboBox {
+                id: sunLightingSelection
+                anchors.horizontalCenter: parent.horizontalCenter
+                width: 200
+                model: [
+                    "No Light",
+                    "Light",
+                    "Light and Shadows"
+                ]
+                currentIndex: 2
+            }
+            Slider {
+                id: sunTimeSlider
+                anchors.horizontalCenter: parent.horizontalCenter
+                opacity: 0.7
+                from: 0; to: 23.99
+                value: 8.5
+                onValueChanged: {
+                    sceneView.sunTime = doubleToDateTime(sunTimeSlider.value)
+                }
+            }
+        }
+    }
+    states: [
+        State {
+            when: sunLightingSelection.currentIndex == 0
+            PropertyChanges {
+                target: sceneView
+                sunLighting: Enums.LightingModeNoLight
+            }
+        },
+        State {
+            when: sunLightingSelection.currentIndex == 1
+            PropertyChanges {
+                target: sceneView
+                sunLighting: Enums.LightingModeLight
+            }
+        },
+        State {
+            when: sunLightingSelection.currentIndex == 2
+            PropertyChanges {
+                target: sceneView
+                sunLighting: Enums.LightingModeLightAndShadows
+            }
+        }
+    ]
+
+    function doubleToDateTime(d) {
+        const remainder = d % 1
+        let minute = remainder * 60
+        let hour = d - remainder
+        if (hour < 10) {
+            hour = "0" + hour
+        }
+        if (minute < 10) {
+            minute = "0" + minute
+        }
+        const dateTime = new Date("2018-08-10T"+hour+":"+minute)
```

Comment:
Couple of suggestions: 

- could just return here
- could use newish ECMAScript 6 string interpolation - https://stackoverflow.com/questions/1408289/how-can-i-do-string-interpolation-in-javascript

```suggestion
        return new Date(`2018-08-10T ${hour} : ${minute}`);
```